lineage/BaumWelch.py

```diff
@@ -145,6 +150,31 @@ def do_M_E_step(tHMMobj, gammas):
         tHMMobj.estimate.E[state_j].estimator(all_cells, all_gammas[:, state_j])
 
 
+def do_M_E_step_atonce(all_tHMMobj, all_gammas):
+    """ perform the M_E step when all the concentrations are given at once. """
+    all_cells = []
```

Comment:
You should have this all work for one state at a time. Collecting everything across states makes it more complicated.

Reply:
Do you mean that the functions should be written for one state or you mean I write a for loop over the state inside this function and do everything for each step at each iteration?

Reply:
I mean you should have a function that concerns itself with just one state. You can then call it N states numbers of times.

Reply:
Thank you.